test/workbox-routing/node/Router.mjs

```diff
@@ -0,0 +1,302 @@
+    });
+
+    it(`should return false and not add a fetch listener the second time addFetchListener() is called`, function() {
+      const spy = sandbox.spy(self, 'addEventListener');
+
+      const router = new Router();
+      const addedFirst = router.addFetchListener();
+      const addedSecond = router.addFetchListener();
+
+      expect(addedFirst).to.be.true;
+      expect(addedSecond).to.be.false;
+      expect(spy.calledOnce).to.be.true;
+    });
+  });
+
+  describe(`registering and unregistering routes`, function() {
+    it(`should throw in dev when registerRoute is called with a parameter that doesn't match the expected interface`, async function() {
+      if (process.env.NODE_ENV === 'production') return this.skip();
+
+      const router = new Router();
+
+      await expectError(
+        () => router.registerRoute({}),
+        'missing-a-method',
+        (error) => {
+          expect(error.details).to.have.property('moduleName').that.eql('workbox-routing');
+          expect(error.details).to.have.property('className').that.eql('Router');
+          expect(error.details).to.have.property('funcName').that.eql('registerRoute');
+          expect(error.details).to.have.property('paramName').that.eql('route');
+        }
+      );
+
+      await expectError(
+        () => router.registerRoute({handle: HANDLER, method: METHOD}),
+        'missing-a-method',
+        (error) => {
+          expect(error.details).to.have.property('moduleName').that.eql('workbox-routing');
+          expect(error.details).to.have.property('className').that.eql('Router');
+          expect(error.details).to.have.property('funcName').that.eql('registerRoute');
+          expect(error.details).to.have.property('paramName').that.eql('route');
+        }
+      );
+
+      await expectError(
+        () => router.registerRoute({match: MATCH, method: METHOD}),
+        'incorrect-type',
+        (error) => {
+          expect(error.details).to.have.property('moduleName').that.eql('workbox-routing');
+          expect(error.details).to.have.property('className').that.eql('Router');
+          expect(error.details).to.have.property('funcName').that.eql('registerRoute');
+          expect(error.details).to.have.property('paramName').that.eql('route');
+        }
+      );
+
+      await expectError(
+        () => router.registerRoute({match: MATCH, method: METHOD, handler: {}}),
+        'missing-a-method',
+        (error) => {
+          expect(error.details).to.have.property('moduleName').that.eql('workbox-routing');
+          expect(error.details).to.have.property('className').that.eql('Router');
+          expect(error.details).to.have.property('funcName').that.eql('registerRoute');
+          expect(error.details).to.have.property('paramName').that.eql('route');
+        }
+      );
+
+      await expectError(
+        () => router.registerRoute({match: MATCH, handle: HANDLER}),
+        'incorrect-type',
```

Comment:
Maybe pull these tests apart into missing-a-method and incorrect-type groups and use the generate-test-variants helper to reduce boilerplate.

Reply:
I've switched to generate-test-variants